src/cloudflare/internal/d1-api.ts

```diff
@@ -52,38 +54,102 @@ type SQLError = {
 
 type ResultsFormat = 'ARRAY_OF_OBJECTS' | 'ROWS_AND_COLUMNS' | 'NONE';
 
+type D1SessionCommitTokenOrConstraint = string;
+type D1SessionCommitToken = string;
+
+// TODO Figure out what header to use.
+const D1_SESSION_COMMIT_TOKEN_HTTP_HEADER = 'x-cf-d1-session-commit-token';
```

Comment:
I guess query parameters are probably better to avoid losing them

Reply:
What do you mean avoid losing them?

The reason I chose a header is to make it easy to use in the REST API queries we offer as well in the same way. Since there we pass params in the body instead of query params.

We could support both query params and body params (in that order) though, and that would be fine.

I don't have strong preference, either way works for me.

Reply:
I think @rozenmd means that it's really easy to accidentally drop request headers when chaining fetch requests (since you usually have to copy them into the new sub-request that you're making).  Query parameters are often more explicit since they're part of the URL and we generally proxy the URL.

I don't have a preference either way!

Reply:
OK, I will switch to query params since there is preference there.

Reply:
FYI, I am going to stick with headers now at least for the communication between the D1 Binding and the eyeball worker, since it's easier to not requiring to parse full body responses to extract the token (even if they are already parsed before returned to users).

For the public API we can do query parameters if needed, and the eyeball can handle both.